pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -191,15 +175,30 @@ func (m *OSDHealthMonitor) removeOSDDeploymentIfSafeToDestroy(outOSDid int) erro
 	return nil
 }
 
-// updateCephStorage updates the CR with deviceclass details
-func (m *OSDHealthMonitor) updateCephStatus(devices []string) {
+// updateCephStorageStatus updates the Storage details in the ceph cluster CR
+func (m *OSDHealthMonitor) updateCephStorageStatus() {
 	cephCluster := cephv1.CephCluster{}
 	cephClusterStorage := cephv1.CephStorage{}
 
+	devices, err := client.GetDeviceClasses(m.context, m.clusterInfo)
+	if err != nil {
+		logger.Errorf("failed to get osd device classes. %v", err)
+		return
+	}
+
 	for _, device := range devices {
 		cephClusterStorage.DeviceClasses = append(cephClusterStorage.DeviceClasses, cephv1.DeviceClasses{Name: device})
 	}
-	err := m.context.Client.Get(m.clusterInfo.Context, m.clusterInfo.NamespacedName(), &cephCluster)
+
+	osdStore, err := m.getOSDStoreStatus()
+	if err != nil {
+		logger.Errorf("failed to get osd store type count. %v", osdStore)
```

Comment:
If this failed, will `osdStore` still be valid? 
```suggestion
		logger.Errorf("failed to get osd store status. %v", err)
```

Reply:
updated the log message. 

It fails only when we failed to list the OSD deployment. I think then we would just return and don't update OSD store status in the spec, keeping the status prior to the failure intact.